clmm/galaxycluster.py

```diff
@@ -27,45 +20,63 @@ class GalaxyCluster():
+        self.unique_id = None
+        self.ra = None
+        self.dec = None
+        self.z = None
+        self.galcat = None
+        if len(args)>0 or len(kargs)>0:
+            self._add_values(*args, **kargs)
+            self._check_types()
+    def _add_values(self, unique_id: str, ra: float, dec: float, z: float,
                  galcat: Table):
-        if isinstance(unique_id, (int, str)): # should unique_id be a float?
-            unique_id = str(unique_id)
+        """Add values for all attributes"""
+        self.unique_id = unique_id
+        self.ra = ra
+        self.dec = dec
+        self.z = z
+        self.galcat = galcat
+        return
+    def _check_types(self):
+        """Check types of all attributes"""
+        if isinstance(self.unique_id, (int, str)): # should unique_id be a float?
+            self.unique_id = str(self.unique_id)
         else:
             raise TypeError(f'unique_id incorrect type: {type(unique_id)}')
         try:
-            ra = float(ra)
+            self.ra = float(self.ra)
         except ValueError:
-            raise TypeError(f'ra incorrect type: {type(ra)}')
+            raise TypeError(f'ra incorrect type: {type(self.ra)}')
         try:
-            dec = float(dec)
+            self.dec = float(self.dec)
         except ValueError:
-            raise TypeError(f'dec incorrect type: {type(dec)}')
+            raise TypeError(f'dec incorrect type: {type(self.dec)}')
         try:
-            z = float(z)
+            self.z = float(self.z)
         except ValueError:
-            raise TypeError(f'z incorrect type: {type(z)}')
-        if not isinstance(galcat, Table):
-            raise TypeError(f'galcat incorrect type: {type(galcat)}')
-
-        if not -360. <= ra <= 360.:
-            raise ValueError(f'ra={ra} not in valid bounds: [-360, 360]')
-        if not -90. <= dec <= 90.:
-            raise ValueError(f'dec={dec} not in valid bounds: [-90, 90]')
-        if z < 0.:
-            raise ValueError(f'z={z} must be greater than 0')
-
-        self.unique_id = unique_id
-        self.ra = ra
-        self.dec = dec
-        self.z = z
-        self.galcat = galcat
+            raise TypeError(f'z incorrect type: {type(self.z)}')
+        if not isinstance(self.galcat, Table):
+            raise TypeError(f'galcat incorrect type: {type(self.galcat)}')
 
+        if not -360. <= self.ra <= 360.:
+            raise ValueError(f'ra={self.ra} not in valid bounds: [-360, 360]')
+        if not -90. <= self.dec <= 90.:
+            raise ValueError(f'dec={self.dec} not in valid bounds: [-90, 90]')
+        if self.z < 0.:
+            raise ValueError(f'z={self.z} must be greater than 0')
+        return
     def save(self, filename, **kwargs):
         """Saves GalaxyCluster object to filename using Pickle"""
         with open(filename, 'wb') as fin:
             pickle.dump(self, fin, **kwargs)
-
+        return
+    def load(filename, **kwargs):
```

Comment:
This looks fine with me, but I'd ping @matthewkirby because I think I had a convo with him awhile ago about the placement of the load fn.